Plots/Satellites/NCL_sat_1.py

```diff
@@ -0,0 +1,191 @@
+   - Transforming coordinates
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/sat_1.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/sat_1_lg.png
+"""
+
+###############################################################################
+# Import packages:
+import xarray as xr
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import numpy as np
+
+import geocat.datafiles as gdf
+import geocat.viz.util as gvutil
+
+###############################################################################
+# Read in data:
+
+# Open a netCDF data file using xarray default engine and load the data into xarrays
+ds = xr.open_dataset(gdf.get("netcdf_files/slp.1963.nc"), decode_times=False)
+
+# Get data from the 24th timestep
+U = ds.slp[24, :, :]
+
+# Translate short values to float values
+U = U.astype('float64')
+
+# Convert Pa to hPa data
+U = U*0.01
+
+# Fix the artifact of not-shown-data around 0 and 360-degree longitudes
+wrap_U = gvutil.xr_add_cyclic_longitudes(U, "lon")
+
+###############################################################################
+
+def findCoordPressureData(coordarr, lat, lon):
+    # Finds pressure at coordinate given lat, lon, and an array of all the coordinates with data
+    # Lat/lon is not *real* lat/lon.... based on scale given by xarray data [90, -90] and [0, 360]
+    # rather than [-90, 90] and [-180, 180]
+    for x in range(len(coordarr)):
+        for y in range(len(coordarr[x])):
+            if coordarr[x][y][0] == lat and coordarr[x][y][1] == lon:
+                return U.data[x][y]
+
+def makeCoordArr():
+    # Returns an array of (lat, lon) coord tuples with the same dimensions as the pressure data
+    coordarr = []
+    for x in np.array(U.lat):
+        temparr = []
+        for y in np.array(U.lon):
+            temparr.append((x,y))
+        coordarr.append(temparr)
+    return np.array(coordarr)
+
+def findLocalMinima(minPressure=980):
+
+    # Create a 2D array of all the coordinates with pressure data 
+    coordarr = makeCoordArr()
+
+    # Set number that a derivative must be less than in order to classify as a "zero"
+    bound = 0.00001
+
+    # Get global gradient of U.data
+    grad = np.gradient(wrap_U.data)
+    arr1 = grad[0]
+    arr2 = grad[1]
+
+    firstzeroes = np.argwhere(arr1<=bound)
+    secondzeroes = np.argwhere(arr2<=bound)
+
+    commonzeroes = []
+
+    # Find zeroes of both gradient arrays
+    for x in secondzeroes:
+        if x in firstzeroes:
+            commonzeroes.append(x)
+
+    minimacoords = []
+
+    # For every common zero in both gradient arrays 
+    for x in commonzeroes:
+
+        xval = x[0]
+        yval = x[1]-1
+
+        # If the gradient value is a "zero", and if the U.data value is less than minPressure
+        if -minPressure <= wrap_U.data[xval][yval] < minPressure:
```

Comment:
Why are you comparing the pressure at a location of a potential low against the positive *and* negative value of `minPressure`? There shouldn't be any negative pressures.

Reply:
This is a mistake, I meant to add this condition to when I was checking to see if arr1 gradient values were within positive and negative bounds, not for checking pressure values.